File: Lab2/src/main/scala/Coffee/Americano.scala
package Coffee

class Americano(intensity: Intensity, protected val mlOfWater: Int)
  extends Coffee(intensity):

  private val coffeeName: String = "Americano"

  def getMlOfWater: Int = mlOfWater

  override def printCoffeeDetails(): Unit =
    println(s"$coffeeName water: $mlOfWater ml")

  def makeAmericano(): Unit =
    println("Making Americano")
    println(s"Intensity set to $coffeeIntensity")
    println(s"Adding $mlOfWater ml of water")

File: lab2/src/main/kotlin/labo2/Task3/Americano.kt
package org.example.labo2.Task3

import org.example.labo2.Task1.Intensity

// Class for Americano that inherits from Coffee
class Americano(
    private var intensityCoffee: Intensity,
    private var mlOfWater: Int,
    private val coffeeName: String = "Americano"
) : Coffee(intensityCoffee, coffeeName) {
    // Inheriting and overriding printCoffeeDetails from Coffee
    public override fun printCoffeeDetails() {
        super.printCoffeeDetails()
        println("Water: $mlOfWater ml")
    }
    // Method to print recipe of Americano
    public fun makeAmericano(): Americano {
        super.makeCoffee()
        println("Adding $mlOfWater ml of water")
        return this
    }
}